src/playercards/AllCardsBag.ttslua

```diff
@@ -147,89 +333,216 @@ function buildSupplementalIndexes()
         end
       end
 
-      -- Add the card to the appropriate class and level indexes
-      local isGuardian = false
-      local isSeeker = false
-      local isMystic = false
-      local isRogue = false
-      local isSurvivor = false
-      local isNeutral = false
-      local upgradeKey
-      -- Excludes signature cards (which have no class or level) and alternate
-      -- ID entries
-      if (cardMetadata.class ~= nil and cardMetadata.level ~= nil) then
-        isGuardian = string.match(cardMetadata.class, "Guardian")
-        isSeeker = string.match(cardMetadata.class, "Seeker")
-        isMystic = string.match(cardMetadata.class, "Mystic")
-        isRogue = string.match(cardMetadata.class, "Rogue")
-        isSurvivor = string.match(cardMetadata.class, "Survivor")
-        isNeutral = string.match(cardMetadata.class, "Neutral")
-        if (cardMetadata.level > 0) then
-          upgradeKey = "-upgrade"
-        else
-          upgradeKey = "-level0"
+      table.insert(allCardsIndex, cardMetadata.id)
+
+      if cardMetadata.level then
+        table.insert(levelIndex[cardMetadata.level], cardMetadata.id)
+      else
+        otherIndex[cardMetadata.id] = true
+      end
+
+      if cardMetadata.class then
+        local classString = cardMetadata.class .. "|"
+        for class in classString:gmatch("[^|]+") do
+          table.insert(classIndex[string.lower(class)], cardMetadata.id)
         end
-        if (isGuardian) then
-          table.insert(classAndLevelIndex["Guardian"..upgradeKey], cardMetadata.id)
+        local classCount = select(2, string.gsub(classString, "|", ""))
+        table.insert(classIndex[classCount], cardMetadata.id)
+      end    
+
+      local cycleName = cardMetadata.cycle
+      if cycleName ~= nil then
+        cycleName = string.lower(cycleName)
+        if string.match(cycleName, "return") then
+          cycleName = string.sub(cycleName, 11)
         end
-        if (isSeeker) then
-          table.insert(classAndLevelIndex["Seeker"..upgradeKey], cardMetadata.id)
+        if cycleName == "the night of the zealot" then
+          cycleName = "core"
         end
-        if (isMystic) then
-          table.insert(classAndLevelIndex["Mystic"..upgradeKey], cardMetadata.id)
+        cycleIndex[cycleName] = cycleIndex[cycleName] or {}
+        table.insert(cycleIndex[cycleName], cardMetadata.id)
+      end
+
+      local bonded = cardMetadata.bonded
+
+      local checkedTraits = {}
```

Comment:
What is `checkedTraits` used for?

Reply:
Uuuuuuh no clue haha, must be a remnant of an older sorting method.  Removed!